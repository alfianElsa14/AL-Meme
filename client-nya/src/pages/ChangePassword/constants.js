export const CHANGE_PASSWORD_USER = 'CHANGE_PASSWORD_USER';