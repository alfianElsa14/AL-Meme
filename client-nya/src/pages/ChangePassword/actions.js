import { CHANGE_PASSWORD_USER } from "./constants";

export const changePasswordUser = (data, navigate) => ({
    type: CHANGE_PASSWORD_USER,
    data,
    navigate
})