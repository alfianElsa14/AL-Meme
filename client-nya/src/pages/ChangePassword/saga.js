import { changePassword } from "@domain/api"
import { call, takeLatest } from "redux-saga/effects"
import Swal from "sweetalert2";
import { CHANGE_PASSWORD_USER } from "./constants";

const Toast = Swal.mixin({
    toast: true,
    position: "top-end",
    showConfirmButton: false,
    timer: 3000,
    timerProgressBar: true,
    didOpen: (toast) => {
        toast.onmouseenter = Swal.stopTimer;
        toast.onmouseleave = Swal.resumeTimer;
    }
});

export function* doChangePasswordUser({ data, navigate }) {
    try {
        const response = yield call(changePassword, data)
        if (response) {
            Toast.fire({
                icon: "success",
                title: "Change Password is successfully"
            });
            yield call(navigate, '/profile')
        }
    } catch (error) {
        console.log(error.response.status);
        if (error.response.status === 400) {
            const errorMessage = error.response.data.message || "Password required";
            Swal.fire(errorMessage);
        } else {
            Swal.fire("failed change password");
        }
    }
}

export function* changePasswordSaga(){
    yield takeLatest(CHANGE_PASSWORD_USER, doChangePasswordUser)
}