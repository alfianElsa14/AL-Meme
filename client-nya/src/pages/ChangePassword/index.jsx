import React, { useState } from 'react'
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

import { changePasswordUser } from './actions';

import classes from './style.module.scss';

function ChangePassword() {
  const dispatch = useDispatch()
  const navigate = useNavigate()
  const [oldPassword, setOldPassword] = useState('')
  const [newPassword, setNewPassword] = useState('')

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(changePasswordUser({ oldPassword, newPassword }, navigate));
  };

  return (
    <div
      data-testid="ChangePasswordContainer"
      className={classes.container}
    >
      <div className={classes.wrapper}>
        <div
          data-testid="ChangePasswordTitle"
          className={classes.title}
        >
          <FormattedMessage id='app_change_password' />
        </div>
        <form onSubmit={handleSubmit}>
          <div
            data-testid="ChangePasswordInput"
            className={classes.inputItem}
          >
            <label htmlFor="oldPassword">
              <FormattedMessage id='app_old_password' /> :
            </label>
            <input
              type="password"
              id="oldPassword"
              name="oldPassword"
              value={oldPassword}
              onChange={(e) => setOldPassword(e.target.value)}
              data-testid="oldPassword"
            />
          </div>
          <div className={classes.inputItem}>
            <label htmlFor="newPassword">
              <FormattedMessage id='app_new_password' /> :
            </label>
            <input
              type="password"
              id="newPassword"
              name="newPassword"
              value={newPassword}
              onChange={(e) => setNewPassword(e.target.value)}
              data-testid="newPassword"
            />
          </div>
          <button type="submit" className={classes.buttonLogin}>
            <FormattedMessage id='app_save' />
          </button>
        </form>
      </div>
    </div>
  )
}

export default ChangePassword