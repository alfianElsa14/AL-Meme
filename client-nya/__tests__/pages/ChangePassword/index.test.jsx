import ChangePassword from "@pages/ChangePassword";
import classes from "@pages/ChangePassword/style.module.scss";

import { fireEvent, render } from "@utils/testHelper";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    useNavigate: () => mockNavigate,
}));

let wrapper;

beforeEach(() => {
    wrapper = render(<ChangePassword />)
})

describe("page change password", () => {
    test('Should render container correctly', () => {
        const { getByTestId } = wrapper;
        const ChangePasswordContainer = getByTestId('ChangePasswordContainer');
        expect(ChangePasswordContainer).toHaveClass(classes.container)
        expect(ChangePasswordContainer).toBeInTheDocument();
    })

    test('Should render title correctly', () => {
        const { getByTestId } = wrapper;
        const ChangePasswordTitle = getByTestId('ChangePasswordTitle');
        expect(ChangePasswordTitle).toHaveClass(classes.title)
        expect(ChangePasswordTitle).toBeInTheDocument();
    })

    test('Should render input item correctly', () => {
        const { getByTestId } = wrapper;
        const ChangePasswordInput = getByTestId('ChangePasswordInput');
        expect(ChangePasswordInput).toHaveClass(classes.inputItem)
        expect(ChangePasswordInput).toBeInTheDocument();
    })

    test('Should handle input old password change', () => {
        const { getByTestId } = wrapper;
        const oldPasswordInput = getByTestId('oldPassword');
        fireEvent.change(oldPasswordInput, { target: { value : '1234567'}})
        expect(oldPasswordInput.value).toBe('1234567')
    })

    test('Should handle input old password change', () => {
        const { getByTestId } = wrapper;
        const newPasswordInput = getByTestId('newPassword');
        fireEvent.change(newPasswordInput, { target: { value : '123456789'}})
        expect(newPasswordInput.value).toBe('123456789')
    })

    test('Should match with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})