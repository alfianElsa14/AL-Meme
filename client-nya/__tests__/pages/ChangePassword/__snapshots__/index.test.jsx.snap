// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page change password Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="ChangePasswordContainer"
      >
        <div
          class="wrapper"
        >
          <div
            class="title"
            data-testid="ChangePasswordTitle"
          >
            Ganti Password
          </div>
          <form>
            <div
              class="inputItem"
              data-testid="ChangePasswordInput"
            >
              <label
                for="oldPassword"
              >
                Password lama
                 :
              </label>
              <input
                data-testid="oldPassword"
                id="oldPassword"
                name="oldPassword"
                type="password"
                value=""
              />
            </div>
            <div
              class="inputItem"
            >
              <label
                for="newPassword"
              >
                Password baru
                 :
              </label>
              <input
                data-testid="newPassword"
                id="newPassword"
                name="newPassword"
                type="password"
                value=""
              />
            </div>
            <button
              class="buttonLogin"
              type="submit"
            >
              Simpan
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="ChangePasswordContainer"
    >
      <div
        class="wrapper"
      >
        <div
          class="title"
          data-testid="ChangePasswordTitle"
        >
          Ganti Password
        </div>
        <form>
          <div
            class="inputItem"
            data-testid="ChangePasswordInput"
          >
            <label
              for="oldPassword"
            >
              Password lama
               :
            </label>
            <input
              data-testid="oldPassword"
              id="oldPassword"
              name="oldPassword"
              type="password"
              value=""
            />
          </div>
          <div
            class="inputItem"
          >
            <label
              for="newPassword"
            >
              Password baru
               :
            </label>
            <input
              data-testid="newPassword"
              id="newPassword"
              name="newPassword"
              type="password"
              value=""
            />
          </div>
          <button
            class="buttonLogin"
            type="submit"
          >
            Simpan
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
